stepper-lib/src/air/fold/utils.rs

```diff
@@ -0,0 +1,150 @@
+                JValue::Array(array) => {
+                    if array.is_empty() {
+                        // skip fold if array is empty
+                        return Ok(None);
+                    }
+                    array.len()
+                }
+                v => return Err(IncompatibleJValueType((*v).clone(), "array")),
+            };
+
+            let foldable = IterableResolvedCall::init(call_result.clone(), len);
+            Box::new(foldable)
+        }
+        Some(AValue::JValueAccumulatorRef(acc)) => {
+            let acc = acc.borrow();
+            if acc.is_empty() {
+                return Ok(None);
+            }
+
+            let call_results = acc.iter().cloned().collect::<Vec<_>>();
+            let foldable = IterableVecResolvedCall::init(call_results);
+            Box::new(foldable)
+        }
+        _ => {
+            return Err(AquamarineError::InstructionError(String::from(
+                "At now, it isn't possible to use fold iterator in other folds",
+            )))
+        }
+    };
+
+    Ok(Some(iterable))
+}
+
+fn handle_instruction_json_path<'ctx>(
+    exec_ctx: &ExecutionCtx<'ctx>,
+    variable: &str,
+    path: &str,
+) -> Result<Option<IterableValue>> {
+    use AquamarineError::JValueAccJsonPathError;
+    use AquamarineError::JValueJsonPathError;
+
+    let iterable: IterableValue = match exec_ctx.data_cache.get(variable) {
+        Some(AValue::JValueRef(variable)) => {
+            use jsonpath_lib::select;
+
+            let jvalues = select(&variable.result, path)
+                .map_err(|e| JValueJsonPathError(variable.result.deref().clone(), path.to_string(), e))?;
+
+            let len = jvalues.len();
+            if len == 0 {
+                return Ok(None);
+            }
+
+            let jvalues = jvalues.into_iter().cloned().collect();
+
+            let tetraplet = SecurityTetraplet {
+                triplet: variable.triplet.clone(),
+                json_path: path.to_string(),
+            };
+
+            let foldable = IterableJsonPathResult::init(jvalues, tetraplet);
+            Box::new(foldable)
+        }
+        Some(AValue::JValueAccumulatorRef(acc)) => {
+            use jsonpath_lib::select_with_iter;
+
+            let acc = acc.borrow();
+            if acc.is_empty() {
+                return Ok(None);
+            }
+
+            let path = path.to_string();
+            let (jvalues, tetraplet_indices) = select_with_iter(acc.iter().map(|v| v.result.deref()), &path)
+                .map_err(|e| JValueAccJsonPathError(acc.clone(), path.clone(), e))?;
+            let jvalues = jvalues.into_iter().cloned().collect();
+            let tetraplets = tetraplet_indices
+                .iter()
+                .map(|&id| &acc[id].triplet)
+                .map(|triplet| SecurityTetraplet {
+                    triplet: triplet.clone(),
+                    json_path: path.clone(),
+                })
+                .collect::<Vec<_>>();
```

Comment:
Maybe use `to_string` as late as possible? `path.clone()` => `path.to_string()`

```suggestion
            let (jvalues, tetraplet_indices) = select_with_iter(acc.iter().map(|v| v.result.deref()), path)
                .map_err(|e| JValueAccJsonPathError(acc.clone(), path.to_string(), e))?;
            let jvalues = jvalues.into_iter().cloned().collect();
            let tetraplets = tetraplet_indices
                .iter()
                .map(|&id| &acc[id].triplet)
                .map(|triplet| SecurityTetraplet {
                    triplet: triplet.clone(),
                    json_path: path.to_string(),
                })
                .collect::<Vec<_>>();
```

Reply:
Yep, good idea. I checked internals a little bit and to_string as fast as clone.